include/net/nrf_cloud_codec.h

```diff
@@ -20,37 +26,530 @@ extern "C" {
  * @{
  */
 
+/** @brief Object types for encoding and decoding */
+enum nrf_cloud_obj_type {
+	NRF_CLOUD_OBJ_TYPE__UNDEFINED,
+
+	NRF_CLOUD_OBJ_TYPE_JSON,
+	NRF_CLOUD_OBJ_TYPE_CBOR,
+
+	NRF_CLOUD_OBJ_TYPE__LAST,
+};
+
+/** @brief Encoded data sources */
+enum nrf_cloud_enc_src {
+	/** Encoded data is not set */
+	NRF_CLOUD_ENC_SRC_NONE,
+	/** Encoded data was set by @ref nrf_cloud_obj_cloud_encode */
+	NRF_CLOUD_ENC_SRC_CLOUD_ENCODED,
+	/** Encoded data was set manually or by @ref NRF_CLOUD_OBJ_PRE_ENC_DEFINE */
+	NRF_CLOUD_ENC_SRC_PRE_ENCODED,
```

Comment:
Just so I understand, would this NRF_CLOUD_ENC_SRC_PRE_ENCODED be used with CBOR data that I encode using the CDDL, for example?

Reply:
it's mostly just a convenience type so that you can set some data in the obj and use the obj to send the data to the cloud... see usage of `NRF_CLOUD_OBJ_PRE_ENC_DEFINE` in  `slm_at_gnss.c`.